### PR TITLE
Implement CS checking based on the `WP_CLI_CS` ruleset

### LINKED ISSUE #31: Adopt and enforce new `WP_CLI_CS` standard
We have a new PHPCS standard for WP-CLI called `WPCliCS` (props @jrfnl). It is part of the `wp-cli/wp-cli-tests` package starting with version v2.1.0.

To adopt & enforce this new standard, the following actions need to be taken for this repository:

- [ ] Create a PR that adds a custom ruleset `phpcs.xml.dist` to the repository

    - Add `phpcs.xml.dist` file
    - Adapt `.distignore` to ignore `phpcs.xml.dist` & `phpunit.xml.dist`
    - Adapt `.gitignore` to ignore `phpunit.xml`, `phpcs.xml` & `.phpcs.xml`
    - Require version `^2.1` of the `wp-cli/wp-cli-tests` as a dev dependency
- [ ] Make any required changes to the code that fail the checks from the above ruleset in separate PRs
- [ ] Merge thre ruleset once all required changes have been processed and merged

A sample PR for a simple repository can be seen here: https://github.com/wp-cli/maintenance-mode-command/pull/3/files

Related https://github.com/wp-cli/wp-cli/issues/5179

### PR DESCRIPTION
**Pull request by thrijith**
Add a PHPCS ruleset using the new `WP_CLI_CS` standard.

Fixes #31

Related https://github.com/wp-cli/wp-cli/issues/5179

### CODE REVIEW

File: src/Capabilities_Command.php [hunk 1]
```diff
@@ -66,38 +64,35 @@ class Capabilities_Command extends WP_CLI_Command {
...
-		else {
+		} else {
 			$capabilities = array_filter( $role_obj->capabilities );
 		}
-		
+
 		$output_caps = array();
 		foreach ( $capabilities as $cap => $grant ) {
-			$output_cap = new StdClass;
+			$output_cap = new StdClass();
```
Comment (schlessera) [inline src/Capabilities_Command.php]: The class name is camelCase here:
```suggestion
			$output_cap = new stdClass();
```

File: src/Capabilities_Command.php [hunk 2]
```diff
@@ -66,38 +64,35 @@ class Capabilities_Command extends WP_CLI_Command {
...
-				} 
-				else {
+				} else {
 					WP_CLI::line( $cap->name );
 				}
 			}
-		}
-		else {
+		} else {
 			$formatter = new \WP_CLI\Formatter( $assoc_args, $this->fields );
```
Comment (schlessera) [inline src/Capabilities_Command.php]: Please import the dependencies at the top and use relative classes/namespaces.
```suggestion
			$formatter = new Formatter( $assoc_args, $this->fields );
```

File: src/Role_Command.php [hunk 3]
```diff
@@ -253,26 +252,28 @@ public function reset( $args, $assoc_args ) {
 
 		self::persistence_check();
 
-		if ( ! \WP_CLI\Utils\get_flag_value( $assoc_args, 'all' ) && empty( $args ) )
-			WP_CLI::error( "Role key not provided, or is invalid." );
+		if ( ! \WP_CLI\Utils\get_flag_value( $assoc_args, 'all' ) && empty( $args ) ) {
```
Comment (schlessera) [inline src/Role_Command.php]: Import `WP_CLI\Utils` at the top.
```suggestion
		if ( ! Utils\get_flag_value( $assoc_args, 'all' ) && empty( $args ) ) {
```

File: src/Role_Command.php [hunk 4]
```diff
@@ -253,26 +252,28 @@ public function reset( $args, $assoc_args ) {
...
-		$all_roles = array_keys( $wp_roles->roles );
+		$all_roles     = array_keys( $wp_roles->roles );
 		$preserve_args = $args;
 
 		// Get our default roles.
-		$default_roles = $preserve = array( 'administrator', 'editor', 'author', 'contributor', 'subscriber' );
-		$before = array();
+		$default_roles = [ 'administrator', 'editor', 'author', 'contributor', 'subscriber' ];
+		$preserve      = [ 'administrator', 'editor', 'author', 'contributor', 'subscriber' ];
+		$before        = array();
```
Comment (schlessera) [inline src/Role_Command.php]: Stick to one notation.
```suggestion
		$before        = [];
```

File: src/Role_Command.php [hunk 5]
```diff
@@ -253,26 +252,28 @@ public function reset( $args, $assoc_args ) {
...
 
 		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'all' ) ) {
```
Comment (schlessera) [inline src/Role_Command.php]:
```suggestion
		if ( Utils\get_flag_value( $assoc_args, 'all' ) ) {
```

File: src/Role_Command.php [hunk 6]
```diff
@@ -334,21 +335,22 @@ public function reset( $args, $assoc_args ) {
...
+				$removed_cap        = array_diff_key( $before[ $role_key ]->capabilities, $after[ $role_key ]->capabilities );
 				$restored_cap_count = count( $restored_cap );
-				$removed_cap_count = count( $removed_cap );
-				$restored_text = ( 1 === $restored_cap_count ) ? '%d capability' : '%d capabilities';
-				$removed_text = ( 1 === $removed_cap_count ) ? '%d capability' : '%d capabilities';
-				$message = "Restored ". $restored_text . " to and removed " . $removed_text . " from '%s' role.";
+				$removed_cap_count  = count( $removed_cap );
+				$restored_text      = ( 1 === $restored_cap_count ) ? '%d capability' : '%d capabilities';
+				$removed_text       = ( 1 === $removed_cap_count ) ? '%d capability' : '%d capabilities';
+				$message            = 'Restored ' . $restored_text . ' to and removed ' . $removed_text . " from '%s' role.";
```
Comment (schlessera) [inline src/Role_Command.php]:
```suggestion
				$message            = "Restored {$restored_text} to and removed {$removed_text} from '%s' role.";
```

File: phpcs.xml.dist [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+			<property name="prefixes" type="array">
+				<element value="WP_CLI\Role"/><!-- Namespaces. -->
+				<element value="wpcli_role"/><!-- Global variables and such. -->
+			</property>
+		</properties>
+	</rule>
+
+	<!-- Exclude existing classes from the prefix rule as it would break BC to prefix them now. -->
+	<rule ref="WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedClassFound">
+		<exclude-pattern>*/src/(Role|Capabilities)?_Command\.php$</exclude-pattern>
```
Comment (jrfnl) [inline phpcs.xml.dist]: Question mark needs to be removed. There is no file called `_Command.php`.

```suggestion
		<exclude-pattern>*/src/(Role|Capabilities)_Command\.php$</exclude-pattern>
```

File: src/Role_Command.php [hunk 8]
```diff
@@ -253,26 +252,28 @@ public function reset( $args, $assoc_args ) {
...
 
 		global $wp_roles;
-		$all_roles = array_keys( $wp_roles->roles );
+		$all_roles     = array_keys( $wp_roles->roles );
 		$preserve_args = $args;
 
 		// Get our default roles.
-		$default_roles = $preserve = array( 'administrator', 'editor', 'author', 'contributor', 'subscriber' );
-		$before = array();
+		$default_roles = [ 'administrator', 'editor', 'author', 'contributor', 'subscriber' ];
```
Comment (jrfnl) [inline src/Role_Command.php]: Would it make sense to declare this array as a class property and use it here and on the line below ?

**Comment by schlessera** [comment]
I still added a few minor changes, @thrijith. As soon as the tests pass, this one can be merged.

**Pull request merged**